modules/common/src/main/scala/surge/kafka/streams/AggregateStateStoreKafkaStreams.scala

```diff
@@ -158,4 +174,32 @@ class AggregateStateStoreKafkaStreams[Agg >: Null](
   private[streams] def getTopology: Future[Topology] = {
     underlyingActor.ask(GetTopology).mapTo[Topology]
   }
+
+  private def registrationCallback(): Try[Any] => Unit = {
+    case Success(_) =>
+      val registrationResult = signalBus.register(
+        control = this,
+        componentName = "state-store-kafka-streams",
+        shutdownSignalPatterns = shutdownSignalPatterns(),
+        restartSignalPatterns = restartSignalPatterns())
+
+      registrationResult.onComplete {
+        case Failure(exception) =>
+          log.error(s"$getClass registration failed", exception)
+        case Success(done) =>
+          log.debug(s"$getClass registration succeeded - ${done.success}")
+      }(system.dispatcher)
+    case Failure(error) =>
+      log.error(s"Unable to register $getClass for supervision", error)
+  }
+
+  private def start(stopped: ControlAck): Future[ControlAck] = {
+    if (stopped.success) {
+      start()
+    } else {
+      Future {
```

Comment:
you can use Future.successful(ControlAck(success = false, error = Some(new RuntimeException(s"Failed to stop $getClass")))

Reply:
done